data/reusables/actions/jobs/section-defining-outputs-for-jobs.md

```diff
@@ -40,3 +34,49 @@ jobs:
 ```
 
 {% endraw %}
+
+{% note %}
+
+**Note:** When using strategy matrix, job outputs will be combined from all jobs inside matrix. The last executed job outputs will override matching outputs from previous jobs within same matrix.
+
+{% endnote %}
+
+{% raw %}
+
+```yaml
+jobs:
+  job1:
+    runs-on: ubuntu-latest
+    outputs:
+      output_1: ${{ steps.gen_output.outputs.output_1 }}
+      output_2: ${{ steps.gen_output.outputs.output_2 }}
+      output_3: ${{ steps.gen_output.outputs.output_3 }}
+    strategy:
+      matrix:
+        job: [1, 2, 3]
+    steps:
+      - name: Generate output
+        id: gen_output
+        run: |
+          job="${{ matrix.job }}"
+          echo "output_${job}=${job}" >> "$GITHUB_OUTPUT"
+          if [[ $job -eq 3 ]]; then
+            # slow down job so it will hopefully finish executing last
+            sleep 10
+            # override output_1 param
+            echo "output_1=${job}" >> "$GITHUB_OUTPUT"
+          fi
+  job2:
+    runs-on: ubuntu-latest
+    needs: [job1]
+    steps:
+      # Will show
+      # {
+      #   "output_1": "3",
+      #   "output_2": "2",
+      #   "output_3": "3"
+      # }
+      - run: echo '${{ toJSON(needs.job1.outputs) }}'
+```
+
+{% endraw %}
```

Comment:
Add a `{% warning %}` block at the bottom with the message about overriding jobs in later steps.  Something like this would work:

```
Actions does not guarantee the order that matrix jobs will run in. Ensure that the output name is unique, otherwise the last matrix job that runs will override the output value.
```

Reply:
Added at the end as a warning.